File: src/main/java/net/gtaun/shoebill/event/vehicle/VehicleSirenStateChangeEvent.java
package net.gtaun.shoebill.event.vehicle;

import net.gtaun.shoebill.object.Player;
import net.gtaun.shoebill.object.Vehicle;

/**
 * Created by marvin on 01.05.15 in project shoebill-api.
 * Copyright (c) 2015 Marvin Haschker. All rights reserved.
 */
public class VehicleSirenStateChangeEvent extends VehicleEvent {

    private Player player;
    private boolean newState;

    public VehicleSirenStateChangeEvent(Vehicle vehicle, Player player, boolean newState) {
        super(vehicle);
        this.player = player;
        this.newState = newState;
    }

    public Player getPlayer() {
        return player;
    }

    public boolean getNewState() {
        return newState;
    }

}

File: src/main/java/net/gtaun/shoebill/event/vehicle/VehicleSirenStateChangeEvent.kt
package net.gtaun.shoebill.event.vehicle

import net.gtaun.shoebill.entities.Player
import net.gtaun.shoebill.entities.Vehicle

/**
 * This event will be called when [vehicle] changes it's siren state.
 *
 * @author Marvin Haschker
 */
class VehicleSirenStateChangeEvent(vehicle: Vehicle, val player: Player, val newState: Boolean) : VehicleEvent(vehicle)
